### ISSUE TITLE
how to set the margin?

### ISSUE THREAD

**Issue by baicwang** [davidsandberg/facenet#225]
Hi,

I have two questions, both about the alignment margin.
1. In the wiki, you mentioned "32 pixels with an image size of 160 pixels corresponds to a margin of 44 pixels with an image size of 182". But 160-32=128, 182-44=138, they are not equal.
2. In align_dataset_mtcnn.py, you did cropping before resizing. I think it should be resizing first. Otherwise, the margin would be resized too.

Thanks very much.

**Comment by JianbangZ** [comment]
@baicwang 
I think this can be optimized too.
I am actually in the middle of changing this and trying out the effect.
I am designing like below:
step 1: only cropping the face and adding 10% margin to four sides, then save to jepg images using mtcnn (no resizing)
step 2: in the training code, I find the smallest side, resize to 180, but maintain the original aspect ratio. This means, some sides might be bigger or smaller
step 3: random crop to 160*160 

The purpose of doing so is to maintain the original aspect ratio. Not sure if this will improve the accuracy a bit but will find out

**Comment by cactuslei** [comment]
@JianbangZ 
Hello, very interested in your cropping method and I wonder will it improve the accuracy?

**Comment by zixia** [comment]
+1

I have the exact same two questions with @baicwang

**Comment by duxingren14** [comment]
+1
I have the same question. What I find is that the we need to add 10% padding on each of the four sides and an extra 5% margin for random cropping. Without enough contest, the performance will be poor.
@JianbangZ  How is your result?

**Comment by JianbangZ** [comment]
@duxingren14 @cactuslei Sorry for the late I have been busy. 
What I have tried is just after detecting the face with MTCNN, instead of direct resize, I maintain the aspect ratio by slightly stretch the sides the the right place. However, I don't see performance improve or decrease. So I guess it just doesn't really matter if the testing image follows the same rule as training dataset.
